lib/flipper/adapters/dalli.rb

```diff
@@ -0,0 +1,75 @@
+require 'dalli'
+
+module Flipper
+  module Adapters
+    # Internal: Adapter that wraps another adapter with the ability to cache
+    # adapter calls in Memcached using the Dalli gem.
+    class Dalli
+      include Flipper::Adapter
+      FeaturesKey = :flipper_features
+
+      #Internal
+      attr_accessor :cache
+
+      #Public: The name of the adapter.
+      attr_accessor :name
+
+      # Public
+      def initialize(adapter, cache, ttl = 0)
+        @adapter = adapter
+        @name = adapter.name
```

Comment:
I'm inclined to leave this as :dalli or :cached if we go that route. I think there is value in this adapter's name being different than the adapter it is wrapping so you can instrument them separately to see cached and uncached performance.